### PR TITLE
Test for incorrect number transactions in a group

### PR DESCRIPTION
**Pull request by pzbitskiy**
## Summary

As it says, e2e test checking incomplete or overflown group

## Test Plan

e2e group_test

### CODE REVIEW

File: test/e2e-go/features/transactions/group_test.go [hunk 1]
```diff
@@ -202,3 +203,80 @@ func TestGroupTransactionsDifferentSizes(t *testing.T) {
 		a.Error(err)
 	}
 }
+
+func TestGroupTransactionsSubmission(t *testing.T) {
+	t.Parallel()
+	a := require.New(t)
+
+	var fixture fixtures.RestClientFixture
+	fixture.Setup(t, filepath.Join("nettemplates", "TwoNodes50EachFuture.json"))
```
Comment (tsachiherman) [inline test/e2e-go/features/transactions/group_test.go]: No need to use future anymore.

File: test/e2e-go/features/transactions/group_test.go [hunk 2]
```diff
@@ -202,3 +203,80 @@ func TestGroupTransactionsDifferentSizes(t *testing.T) {
...
+	defer fixture.Shutdown()
+
+	client := fixture.LibGoalClient
+	accountList, err := fixture.GetWalletsSortedByBalance()
+	a.NoError(err)
+	account0 := accountList[0].Address
+	goodGroupSizes := []int{1, 2, 16}
+	exceedGroupSize := 17
```
Comment (tsachiherman) [inline test/e2e-go/features/transactions/group_test.go]: avoid using the number 16 and 17 directly. instead, query the consensus params and use the magic numbers from there.

File: test/e2e-go/features/transactions/group_test.go (hunk 1 above)
Comment (pzbitskiy) [reply test/e2e-go/features/transactions/group_test.go]: Rebased and fixed

File: test/e2e-go/features/transactions/group_test.go [hunk 3]
```diff
@@ -202,3 +203,80 @@ func TestGroupTransactionsDifferentSizes(t *testing.T) {
...
+			txns = append(txns, txn)
+		}
+
+		// compute gid
+		gid, err := client.GroupID(txns)
+		a.NoError(err)
+
+		// fill in gid and sign and keep track of txids
+		var stxns []transactions.SignedTxn
+		txids := make(map[string]string)
```
Comment (tsachiherman) [inline test/e2e-go/features/transactions/group_test.go]: why do you need to use `txids` ? I don't see that you're using it.

Comment (pzbitskiy) [reply test/e2e-go/features/transactions/group_test.go]: copy-paste of signing code from above, fixed in both places.

File: test/e2e-go/features/transactions/group_test.go (hunk 2 above)
Comment (pzbitskiy) [reply test/e2e-go/features/transactions/group_test.go]: done

**Pull request merged**